src/screens/Asl/components/AslTable/index.js

```diff
@@ -0,0 +1,236 @@
+          }
+          if (a[sortConfig.key] > b[sortConfig.key]) {
+            return sortConfig.direction === "ascending" ? 1 : -1;
+          }
+          return 0;
+        });
+      }
+      return sortableItems;
+    }, [items, sortConfig]);
+  
+    const requestSort = key => {
+      let direction = "ascending";
+      if (sortConfig && 
+        sortConfig.key === key && 
+        sortConfig.direction === 'ascending') {
+            direction = 'descending';
+        }
+      setSortConfig({ key, direction });
+    };
+  
+    return { items:sortedItems, requestSort, sortConfig};
+};
+
+
+
+const AslTable = props => {
+    const ONE_PAGE_NUM = 50 // one page will have at most 50 glossaries
+    const { items, requestSort, sortConfig } = useSortableData(props.words);
+    const [pageNumber, setPageNumber] = useState(1); // the page number is at first 1
+    const [jumpNumber, setJumpNumber] = useState(1);
+    const [length, setLength] = useState(0); // the length of filtered items is set to 0
+    const [search, setSearch] = useState(''); // search text is at first empty
+    const [onePage, setOnePage] = useState([]);
+
+    const apiInstance = axios.create({
+      baseURL: 'https://ct-dev.ncsa.illinois.edu',
+      timeout: 1000,
+    });
+
+    useEffect(() => {
+      const index = (pageNumber-1) * ONE_PAGE_NUM;
+      setOnePage(items.filter(item => item.term.toLowerCase().includes(search.toLowerCase())).slice(index, index + ONE_PAGE_NUM));
+    }, [pageNumber]); // we set page after pagenumber changed
+
+    useEffect(() => {
+      const filtered = items.filter(item => item.term.toLowerCase().includes(search.toLowerCase()));
+      setLength(filtered.length);
+      setPageNumber(1);
+      setOnePage(filtered.slice(0, 50));
+    }, [search])
+
+    useEffect(() => {
+      setLength(items.length);
+      setSearch('');
+      setPageNumber(1);
+      setOnePage(items.slice(0, 50));
+    }, [items])
+
+    const getClassNamesFor = name => {
+        if (!sortConfig) {
+          return;
+        }
+        return sortConfig.key === name ? sortConfig.direction : undefined;
+    };
+    
+    const handleJump = () => {
+      if (jumpNumber >= 1 && jumpNumber*ONE_PAGE_NUM < length) {
+        // pagenumber is valid
+        setPageNumber(jumpNumber);
+      } else {
+        alert("please enter a valid page number!");
+      }
+    }
+
+    const numberToCate = (num) => {
+      if (num === 1) {
+        return 'Sign'
+      }
+      if (num === 2) {
+        return 'Definition'
+      }
+      return 'Example'
+    }
+
+    const getVideoLink = (source, uniqueASLIdentifier) => {
+      if (source === 'ASLCORE') {
+        return `https://ct-dev.ncsa.illinois.edu/data/aslvideos/aslcore/original/${uniqueASLIdentifier}.mp4`;
+      }
+      return `https://ct-dev.ncsa.illinois.edu/data/aslvideos/deaftec/original/${uniqueASLIdentifier}.mp4`;
+    }
+
+    const handlePrevPage = () => {
+      setPageNumber(pageNumber-1);
```

Comment:
Should this need to be bounded? e.g. prevent pageNumber from becoming negative

Reply:
I did so by disable the button when it reach its bounds. The prev button will be disabled when page number is 1, and the next button will be disabled when page number is MAX PAGE NUMBER

Reply:
Hmm, that will work but is not best practice - you're putting logic into the UI. page is part of your view model. What if code allowed rows to be deleted? What if there's a new way to update the page number (e.g. by speech).Best practice would be setPageNumber does not trust its input (so bounds the new suggested value to have a min/max). Also later use of pageNumber might do a sanity check - i.e. at least not crash with an exception if the page number was too large for the data

Reply:
What if you later remember the page number but then reloaded the data from the server, and the new dataset was smaller.

Reply:
OK, I will fix that.